src/platforms/ctxlink/platform.c

```diff
@@ -139,25 +145,11 @@ void wifi_init(void)
 	app_initialize();
 }
 
-void platform_adc_read(void)
-{
-	adc_set_regular_sequence(ADC1, 1, &adc_channels[CTXLINK_ADC_BATTERY]);
-	adc_start_conversion_regular(ADC1);
-	/* Wait for end of conversion. */
-	while (!adc_eoc(ADC1))
-		continue;
-	input_voltages[CTXLINK_ADC_BATTERY] = ((adc_read_regular(ADC1) * 16U) / 10U);
-	adc_set_regular_sequence(ADC1, 1, &adc_channels[CTXLINK_ADC_TARGET]);
-	adc_start_conversion_regular(ADC1);
-	/* Wait for end of conversion. */
-	while (!adc_eoc(ADC1))
-		continue;
-	input_voltages[CTXLINK_ADC_TARGET] = (adc_read_regular(ADC1) * 16U) / 10U;
-}
-
 uint32_t platform_target_voltage_sense(void)
 {
-	return input_voltages[CTXLINK_ADC_TARGET];
+	uint32_t voltage;
+	__atomic_load(&input_voltages[CTXLINK_ADC_TARGET], &voltage, __ATOMIC_RELAXED);
```

Comment:
Please include `<stdatomic.h>` and use the [non-underscored](https://en.cppreference.com/w/c/atomic/atomic_load) version of this function which is standardised. It is called in a very similar way, but does not depend on compiler specifics. In this context it would be used as something like:
```c
return atomic_load(input_voltages[CTXLINK_ADC_TARGET], memory_order_relaxed);
```

Reply:
Nice catches all around, I had forgotten how to use 'atomics' and research took me down the wrong path.

Reply:
All good - nice job with the cleanup/fixups!